Composer/packages/client/src/recoilModel/dispatchers/provision.ts

```
@@ -0,0 +1,171 @@
+      description: formatMessage('{msg}', { msg: value }),
+      type: 'pending',
+    };
+  };
+  const getProvisionSuccessNotification = (value: string): CardProps => {
+    return {
+      title: formatMessage('Provision success'),
+      description: formatMessage('{msg}', { msg: value }),
+      type: 'success',
+    };
+  };
+  const getProvisionFailureNotification = (value: string): CardProps => {
+    return {
+      title: formatMessage('Provision failure'),
+      description: formatMessage('{msg}', { msg: value }),
+      type: 'error',
+    };
+  };
+
+  const provisionToTarget = useRecoilCallback(
+    (callbackHelpers: CallbackInterface) => async (
+      config: any,
+      type: string,
+      projectId: string,
+      armToken = '',
+      graphToken = ''
+    ) => {
+      try {
+        console.log('graph token is: ', graphToken);
+        const result = await httpClient.post(`/provision/${projectId}/${type}`, config, {
+          headers: { Authorization: `Bearer ${armToken}`, graphtoken: graphToken },
+        });
+        console.log(result.data);
+        const notification = createNotification(getProvisionPendingNotification(result.data.message));
+        addNotificationInternal(callbackHelpers, notification);
+        // update provision status
+        callbackHelpers.set(provisionStatusState(projectId), (provisionStatus) => {
+          const newStat = {
+            ...provisionStatus,
+            [result.data.processName]: {
+              ...result.data,
+              notificationId: notification.id,
+            },
+          };
+          console.log(newStat);
+          return newStat;
+        });
+
+        // get provision status
+        updateProvisionStatus(
+          callbackHelpers,
+          result.data.id,
+          projectId,
+          result.data.processName,
+          type,
+          notification.id
+        );
+      } catch (error) {
+        console.log(error.response.data);

```

Do we need to trigger a notification instead failed silently?